### DISCUSSION TITLE
How to rerun the exact alr build commands in GnatStudio

### DISCUSSION THREAD

**Discussion by pjljvandelaar** [alire-project/alire#993] [Q&A]
Hi Alire developers,

Before I start gnatstudio, I invoke
`alr printenv --powershell | Invoke-Expression` to set the environment right.
gnatstudio is now able to correctly open my alire project.

Gnatstudio has build, build & run, and debug & run functionality.
Yet the command there are not incorrespondance with the alire settings.
E.g. I get the following build command
```
gprbuild -d -PC:\path\to\dependency_graph_extractor_0.1.0_71bc61cd\dependency_graph_extractor.gpr -XLIBRARY_TYPE=static -XLANGKIT_SUPPORT_LIBRARY_TYPE=static -XBUILD_MODE=dev -XLANGKIT_SUPPORT_BUILD_MODE=dev -XGNATCOLL_BUILD_MODE=PROD -XXMLADA_BUILD_MODE=Production -XGPR_BUILD=production -XGNATCOLL_OS=windows -XGNATCOLL_MADVISE=yes -XGNATCOLL_MMAP=yes -XLIBADALANG_WARNINGS=false -XLIBADALANG_LIBRARY_TYPE=static -XGNATCOLL_ICONV_OPT=-liconv -XGNATCOLL_VERSION=22.0.0 -XPROCESSORS=0 -XOBJDIR=. -XBUILD=PROD -s -k
```
but this is certainly not equal to the build command of alr (who lacks at least the "-s").

Is there an option to get the full build command that alire uses?
If not, should such functionality be added? Or am I missing something?

Thanks in advance for your answer!

Pierre

**Comment by Fabien-Chouteau** [comment]
Hi @pjljvandelaar,

GNAT Studio (GS) is not "Alire aware". So GS will run `gprbuild` directly and not `alr build`, the default options for gprbuild are different in GS and Alire, but you can change the options in GS settings.

Note that there is an `alr edit` command to start GNAT studio from Alire.

**Reply by pjljvandelaar** [reply]
Will experiment some more with GS and its settings!

Thanks for pointing me to `alr edit`. 
I was using `alr printenv --powershell | Invoke_Expression` + `gnatstudio`...

**Reply by pjljvandelaar** [reply]
Where to find the settings you refer to?
I don't think you mean:
* Build > Settings > Toolchain gives a pop up with two paths (complier/tools) to configure.

Do you mean the following preferences?
* Edit > Preferences > Build Targets > Project > Build Main
And should I just hard code the build command in there?

**Reply by Fabien-Chouteau** [reply]
I usually use the `F4 `shortcut to compile in GNAT Studio, this opens a window where I can change some of the switches.
For instance, there is a `Recompile if switches changed` tick box.

**Comment by mosteo** [comment]
Also, once you have set up the same environment, you can see the exact gprbuild invocation used by `alr` among the logs in the verbose debug level (single `-v`, e.g., `alr -v build`)

**Reply by pjljvandelaar** [reply]
As a user, I would like a more directed command since `-v` creates a lot more messages than the invocations in the different build/compile/link steps.
But for now it helps ;-)
